src/modules/navigator/mission_block.cpp

```diff
@@ -201,10 +201,12 @@ MissionBlock::is_mission_item_reached()
 			}
 
 			/* require only altitude for takeoff for multicopter */
-			if (_navigator->get_global_position()->alt >
-			    altitude_amsl - altitude_acceptance_radius) {
-				_waypoint_position_reached = true;
-			}
+			const auto global_altitude_acceptable =
+				_navigator->get_global_position()->alt > altitude_amsl - altitude_acceptance_radius;
+			const auto local_altitude_acceptable =
```

Comment:
I'm not sure that this is safe in general. As a first step what about only checking local acceptable if `_mission_item.altitude_is_relative`?

Reply:
You're probably right. This implementation is very crude and not properly tested, just kind of a "what I had to do to be able to fly". I'll check out the altitude is relative thing.

Reply:
When I'm flying SITL, pushing pose data in via the "external vision" input and setting EKF2_AID_MASK to 24 (vision only, no gps), so that my local estimate is valid but global invalid (checked with `ekf2 status`), it seems that `_mission_item.altitude_is_relative` is false.

Reply:
I wanted to bring up the same concern, the things that come to my mind to make sure would be:
- a local origin that doesn't have z component 0 where the drone is currently standing (we should make that a default in SITL to catch these bugs early on)
- having GPS and loosing it again in which case something in the setup e.g. ground station might still be sending absolute altitude